util/codegen/codegen_test.go

```diff
@@ -174,3 +179,103 @@ func TestGenericContainsPointers(t *testing.T) {
 		})
 	}
 }
+
+func TestAssertStructUnchanged(t *testing.T) {
+	type args struct {
+		t      *types.Struct
+		tname  string
+		params *types.TypeParamList
+		ctx    string
+		it     *ImportTracker
+	}
+	cfg := &packages.Config{
+		Mode:  packages.NeedTypes | packages.NeedTypesInfo | packages.NeedSyntax | packages.NeedName,
+		Tests: true,
+	}
+
+	rootDir := "./codegen_test_pkgs"
+	pkg_map := make(map[string]*types.Package)
+
+	// load packages recursively
+	err := filepath.Walk(rootDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
Can you please set `cfg.Dir` and use `packages.Load(cfg, "./...")` instead?
That's a bit cleaner, and each package will also be loaded only once, with a single representation of each loaded type (e.g., T1, T2, and T3) in memory. Less error-prone.

Reply:
Changed to self-contained structs, the `packages.load` has been removed.